2021 Annual Report: SIG Docs (#6468)
kubernetes/community: sig-docs/annual-report-2021.md

> 1. What work did the SIG do this year that should be highlighted?

Reviewer: any milestones hit or really amazing metrics to share about how much work y'all accomplished? example: page views, PRs reviewed, etc?

Reviewer: did you work with other groups on thorny problems? examples of good collaboration, etc?

Author: I'll be highlighting the ongoing efforts for Dockershim that we flagged off last year in December.

Author: cc: @a-mccarthy for the page views/PRs reviewed bit, if you could please pull in the numbers?

>    - SIG Docs put meaningful effort into growing its contributor and reviewer base in 2021, introducing a shadow program for PR Wrangling as well as dedicating more time to being active via our Slack community channel. This is an ongoing effort to grow our contributor base to a stable number, alongside easing the burden on our small group of reviewers and approvers.

Reviewer: link to the shadow program docs

Author: We don't have a doc per se, Paris. @bradtopol please correct me if I'm mistaken here. We announced it during one of our meetings and asked them to reach out to Brad, if interested. The PR Wranglers list then gets updated with the shadows

Reviewer 2: We can link to the PR Wrangling page itself that has the shadow listing next to each wrangler for the week

Reviewer 3: Is that kind of detail right for an annual report? It'd be better to have a doc about the shadow program generally but right now we don't.

I don't think it's obvious from the PR wrangler list how you become a shadow, which I think is the key thing Paris would want readers to be able to discover.

(instead of adding a link: how about we open and prioritize an issue about adding those contributor docs)?

Reviewer 2: Issue created here: https://github.com/kubernetes/website/issues/31956

Author: Added issue mentioned by @natalisucks , hence resolving comment

>    - Alongside growing our contributor base, SIG Docs also worked on a leadership transition strategy to bring community members into leadership roles. Via a specialized six month mentorship program expertly led by Steering Committee member Paris Pittman, SIG Docs was able to grow its leadership cohort for the main SIG, as well as some of its subgroups, adding new co-chairs and tech leads.

Reviewer: include a link to the mailing list entries with Jim asking for help and then the announcement email

> 2. What initiatives are you working on that aren't being tracked in KEPs?

Reviewer: would another one be the consistent maintenance of the docset and strategies therein on how to make that happen? what about some other initiatives that I see on your agendas? do you have project board for the group or any subprojects?

Author: We don't have a project board, to the best of my knowledge. Ccing the other leads/chairs on here so that they can provide further insight.
@jimangel @sftim @onlydole @kbhawkey @reylejano @a-mccarthy @natalisucks @tengqm

Reviewer 4: There is a project board for the blog subproject
https://github.com/kubernetes/website/projects/11

>    - Localization Subproject: SIG Docs is working on formalizing the localization work that has been ongoing for some time, with appointed leads of this initiative as well as recognizing the contributions of various community members across the different languages the Kubernetes website has been translated into. This subproject will be finalized by Q1 2022, with all active localizations informed and updated.

Reviewer: add any and all applicable links here - issues and proposal doc etc

Author: Cc: @natalisucks @a-mccarthy @bradtopol since I know y'all have been working towards this.

Reviewer 2: Issue created here: https://github.com/kubernetes/website/issues/31955

Author: Added issue mentioned by @natalisucks, hence resolving comment

>    - New Contributor Ambassador Program: As a continuation of our push to grow the SIG Docs contributor base, we're working on a specalized role that aims to support new and would-be contributors get up to speed with our processes and workflows. This role would be capped at six months for it to be shared amongst the community, with this feeding into a possible reviewer funnel as contributors get more comfortable with providing feedback to others.

Reviewer: same here - any links that you can point folks too, even if you have to create the issue now as a placeholder that would be fine. do you want to shoutout anyone doing this work?

Author: Will create the placeholder issue & link here.

Author: Created issue and linked in latest commit.

3. KEP work in 2021 (continuous and does not target any release):

<!--
In future, this will be generated from kubernetes/enhancements kep.yaml files
1. with SIG as owning-sig or in participating-sigs
2. listing 1.x, 1.y, or 1.z in milestones or in latest-milestone
-->

   - [1326 - Doc policies for third party content](https://git.k8s.io/enhancements/keps/sig-docs/1326-third-party-content-in-docs/README.md)

## Project health

1. What areas and/or subprojects does your group need the most help with?

>    Any areas with 2 or fewer OWNERs? (link to more details)

Reviewer 3: I'd call out the blog subproject as short on resources / attention. A lot of the publication that's happening is depending on a single editor (me), and the other editors are perhaps even more stretched with other Kubernetes contributions and SIG involvement(s).

Reviewer: +1 to @sftim - I see the update, too. can you add that exactly detail that Tim mentioned above - one editor and looking for more. this is very important if folks want to continue to have a good SLA for their blog posts.

>    - Require more contributors to form a stable pool for SIG Docs

Reviewer: new contributors? members? reviewers? all?

Reviewer: PR wranglers? 
get really detailed here and include links to any membership documentation, etc

Reviewer: did y'all run dim's maintainer audit tooling?

Reviewer 3: maintainer audit tooling? (I don't think I know what that is)

   - Require guidelines on how we could ramp up folks nominated to the New Contributor Ambassador role specific to SIG Docs.

2. What metrics/community health stats does your group care about and/or measure?

>    - SIG Docs has a [dashboard](https://datastudio.google.com/u/0/reporting/fede2672-b2fd-402a-91d2-7473bdb10f04/page/567IC) available with site anyaltics. Some highlights include: 

Author: Addressed in the latest commit.

>    - SIG Docs has a [dashboard](https://datastudio.google.com/u/0/reporting/fede2672-b2fd-402a-91d2-7473bdb10f04/page/567IC) available with site analytics. Some highlights include: 

Reviewer: do y'all do anything with this data? example: make sure these pages are always updated and fresh?

Reviewer 3: I don't think we do anything with that. I might be wrong though.

       - 2021 Pages views: 111,565,437
       - Top pages for 2021 (excluding the home page): 
           - https://kubernetes.io/docs/reference/kubectl/cheatsheet/
           - https://kubernetes.io/docs/concepts/services-networking/service/
           - https://kubernetes.io/docs/concepts/services-networking/ingress/

>    - PR velocity and open PR age is tracked in [Devstat](https://k8s.devstats.cncf.io/d/25/open-pr-age-by-repository-group?orgId=1&var-period=q&var-repogroup_name=SIG%20Docs&var-kind_name=All). We aim to have < 100 open PRs for the English localization, and will take steps as needed if we see the figure climbing much above that. For example, based on 12/31/2020 data, we have: Average number of opened PRs 368 with the median opened PRs age 5 days 21 hours 26 minutes 42 seconds

Reviewer: this is good info: We aim to have < 100 open PRs for the English localization, and will take steps as needed if we see the figure climbing much above that.

3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
   to activities or programs that provide useful context or allow easy participation?

>    - It's updated to the best of our knowledge.

Reviewer: ```suggestion
   - [It's updated](https://kubernetes.io/docs/contribute/) to the best of our knowledge.
```

4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
   does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?

>    - We have a [well-documented guide](https://kubernetes.io/docs/contribute/#:~:text=Anyone%20can%20open%20an%20issue,the%20CNCF%20Contributor%20License%20Agreement.) that details how folks can get started and the various ways in which they can scale the contributor ladder.

Reviewer: ```suggestion
   - We have a [well-documented guide](https://kubernetes.io/docs/contribute/) that details how folks can get started and the various ways in which they can scale the contributor ladder.
```

>    - We actively assign good-first-issue labels to issues that we think early-lifecycle contributors could begin with. During KubeCons we also do introduction presentations and mentoring sessions to try and funnel contributors in to our SIG.

Reviewer: have any links to those sessions?

Author: We didn't have any this past year. I combed through Youtube but only found Jim's, Zach's and Seokho's presentations at previous KubeCons. Would it be worthwhile listing them here? Jim's presentation is already listed below.

Reviewer 3: There's also https://www.youtube.com/watch?v=pprMgmNzDcw and https://www.youtube.com/watch?v=vz_5EdhHrmI

Author: Added only the first one, since the second one is very generic and talks about docs in general.

5. Does the group have contributors from multiple companies/affiliations?

>    - [98 companies](https://k8s.devstats.cncf.io/d/8/company-statistics-by-repository-group?orgId=1&var-period=y&var-metric=contributions&var-repogroup_name=SIG%20Docs&var-repo_name=kubernetes%2Fkubernetes&var-companies=All&from=1609455600000&to=1639350000000)

Reviewer: I would call out the companies that do the top 50% and also make a statement about the independents in your group - that is a huge number and speaks to the types of support that your contributor base gets when they contribute

Author: Added in last commit

6. Are there ways end users/companies can contribute that they currently are not?
   If one of those ways is more full time support, what would they work on and why?

>    - We're in the process of formalizing the localization subproject. We aim for it to, potentially, provide more avenues for contribution. More details around leadership for the localization supbproject can be found in [this message](https://groups.google.com/a/kubernetes.io/g/dev/c/SP6weMvx3wg/m/l8LAL-OFCQAJ) sent to dev@kubernetes.io 

Reviewer: is there a statement we can make in this section about the importance of localization and why end users should care?

Author: Yes, I will draft something.

## Membership

- Primary slack channel member count: 1876
- Primary mailing list member count: 453
- Primary meeting attendee count (estimated, if needed): 10
- Primary meeting participant count (estimated, if needed): 6
- Unique reviewers for SIG-owned packages: 76 <!-- in future, this will be generated from OWNERS files referenced from subprojects, expanded with OWNERS_ALIASES files -->

> - Unique approvers for SIG-owned packages: 93<!-- in future, this will be generated from OWNERS files referenced from subprojects, expanded with OWNERS_ALIASES files -->

Reviewer 3: Does this make sense for SIG Docs?

Author: Tbh, I don't think. But @parispittman would you please be able to comment if this is actually relevant for non-code owning SIGs?

   - New Contributor Ambassador Program: As a continuation of our push to grow the SIG Docs contributor base, we're working on a specalized role that aims to support new and would-be contributors get up to speed with our processes and workflows. This role would be capped at six months for it to be shared amongst the community, with this feeding into a possible reviewer funnel as contributors get more comfortable with providing feedback to others. The issue tracking the formalization and documentation of this role can be viewed [here](https://github.com/kubernetes/website/issues/31946).

3. KEP work in 2021 (continuous and does not target any release):

<!--
In future, this will be generated from kubernetes/enhancements kep.yaml files
1. with SIG as owning-sig or in participating-sigs
2. listing 1.x, 1.y, or 1.z in milestones or in latest-milestone
-->

   - [1326 - Doc policies for third party content](https://git.k8s.io/enhancements/keps/sig-docs/1326-third-party-content-in-docs/README.md)

## Project health

1. What areas and/or subprojects does your group need the most help with?
   Any areas with 2 or fewer OWNERs? (link to more details)

>    - Require more contributors to form a stable pool for the blog subproject, in particular and SIG Docs, generally.

Reviewer 3: ```suggestion
   - The blog subproject is particularly short on resources and attention. At the moment a very
     small pool of active editors are the constraint / most critical resource for article publication. One editor
     is involved in the majority of published articles;  other editors are perhaps even more stretched with other
     Kubernetes contributions and involvement with other SIGs.
     - Whilst there is work under way to move other people along the contribution ladder, the blog subproject
       remains constrained.
   - SIG Docs would welcome more contributors to form a stable pool for the blog subproject in particular and
     for SIG Docs, generally.
```

   - Require guidelines on how we could ramp up folks nominated to the New Contributor Ambassador role specific to SIG Docs. 

> 

Reviewer 3: I hope this is written from a sufficiently neutral point of view
```suggestion
    - One localization (Ukrainian) is primarily worked on by people based in Ukraine,
      where the ongoing and intensifying conflict creates challenges that take priority over
      open source contribution.
```

I used “conflict” rather than “invasion” because I imagine that the CNCF would rather take a neutral stance on the Russian invasion.

2. listing 1.x, 1.y, or 1.z in milestones or in latest-milestone
-->

   - [1326 - Doc policies for third party content](https://git.k8s.io/enhancements/keps/sig-docs/1326-third-party-content-in-docs/README.md)

## Project health

1. What areas and/or subprojects does your group need the most help with?
   Any areas with 2 or fewer OWNERs? (link to more details)

   - The blog subproject is particularly short on resources and attention. At the moment a very
     small pool of active editors are the constraint / most critical resource for article publication. One editor
     is involved in the majority of published articles;  other editors are perhaps even more stretched with other
     Kubernetes contributions and involvement with other SIGs.
     - Whilst there is work under way to move other people along the contribution ladder, the blog subproject
       remains constrained.
   - SIG Docs would welcome more contributors to form a stable pool for the blog subproject in particular and
     for SIG Docs, generally.
   - Require guidelines on how we could ramp up folks nominated to the New Contributor Ambassador role specific to SIG Docs. 

>     - One localization (Ukrainian) is primarily worked on by people based in Ukraine,
>       where the ongoing and intensifying conflict creates challenges that take priority over
>       open source contribution.

Reviewer 3: Oops, my bad formatting needs a fixup
```suggestion
   - One localization (Ukrainian) is primarily worked on by people based in Ukraine,
     where the ongoing and intensifying conflict creates challenges that take priority over
     open source contribution.
```
